examples/pytorch/infograph/semisupervised.py

```diff
@@ -0,0 +1,222 @@
+import numpy as np
+import torch as th
+import torch.nn.functional as F
+import dgl
+from dgl.dataloading import GraphDataLoader
+from qm9_v2 import QM9DatasetV2
+from model import InfoGraphS
+import argparse
+
+
+def argument():
+    parser = argparse.ArgumentParser(description='InfoGraphS')
+
+    # data source params
+    parser.add_argument('--target', type=str, default='mu', help='Choose regression task')
+    parser.add_argument('--train_num', type=int, default=5000, help='Size of training set')
+
+    # training params
+    parser.add_argument('--gpu', type=int, default=-1, help='GPU index, default:-1, using CPU.')
+    parser.add_argument('--epochs', type=int, default=200, help='Training epochs.')
+    parser.add_argument('--batch_size', type=int, default=20, help='Training batch size.')
+    parser.add_argument('--val_batch_size', type=int, default=100, help='Validation batch size.')
+
+    parser.add_argument('--lr', type=float, default=0.001, help='Learning rate.')
+    parser.add_argument('--wd', type=float, default=0, help='Weight decay.')
+
+    # model params
+    parser.add_argument('--hid_dim', type=int, default=64, help='Hidden layer dimensionality')
+    parser.add_argument('--reg', type=float, default=0.001, help='Regularization coefficient')
+
+    args = parser.parse_args()
+
+    # check cuda
+    if args.gpu != -1 and th.cuda.is_available():
+        args.device = 'cuda:{}'.format(args.gpu)
+    else:
+        args.device = 'cpu'
+
+    return args
+
+
+def collate(samples):
+    ''' collate function for building graph dataloader '''
+
+    # generate batched graphs and labels
+    graphs, targets = map(list, zip(*samples))
+    batched_graph = dgl.batch(graphs)
+    batched_targets = th.Tensor(targets)
+    
+    n_graphs = len(graphs)
+    graph_id = th.arange(n_graphs)
+    graph_id = dgl.broadcast_nodes(batched_graph, graph_id)
+    
+    batched_graph.ndata['graph_id'] = graph_id
+    
+    return batched_graph, batched_targets
+
+
+if __name__ == '__main__':
+
+    # Step 1: Prepare graph data   ===================================== #
+    args = argument()
+    label_keys = [args.target]
+    print(args)
+
+    dataset = QM9DatasetV2(label_keys)
+    dataset.to_dense()
+
+    graphs = dataset.graphs
+
+    # Train/Val/Test Splitting
+    N = len(graphs)
+    all_idx = np.arange(N)
+    np.random.shuffle(all_idx)
+
+    val_num = 10000
+    test_num = 10000
+
+    val_idx = all_idx[:val_num]
+    test_idx = all_idx[val_num : val_num + test_num]
+    train_idx = all_idx[val_num + test_num : val_num + test_num + args.train_num]
+
+    train_data = [dataset[i] for i in train_idx]
+    val_data = [dataset[i] for i in val_idx]
+    test_data = [dataset[i] for i in test_idx]
+
+    unsup_idx = all_idx[val_num + test_num:]
+    unsup_data = [dataset[i] for i in unsup_idx]
```

Comment:
You can use [this API](https://docs.dgl.ai/api/python/dgl.data.html#dgl.data.utils.Subset) for `train_data`, `val_data`, `test_data` and `unsup_data`.